### ISSUE TITLE
body-pix build size is too large

### ISSUE THREAD

**Issue by satyamgaur** [tensorflow/tfjs#7517]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOs 12.6.3 | M1 Chip
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow.js installed from (npm or script link): yarn
- TensorFlow.js version: "@tensorflow-models/body-pix": "2.2.0", "@tensorflow/tfjs": "3.20.0",

**Describe the problem**
We are currently using only the `load` function of the body-pix module, but still the file size is coming around 27mb (5.74 mb gzipped) as if tree-shaking is not at all happening. We have a very big repo, but no other package/module is leading to such a big file size, what could be the issue behind this? And how we can counter this? Also what are some recommended webpack configs for this?

**Comment by gaikwadrahul8** [comment]
Hi, @satyamgaur 

It seems like body-pix package is deprecated in favor of the new [body-segmentation package](https://github.com/tensorflow/tfjs-models/blob/master/body-segmentation). We have added note [here](https://github.com/tensorflow/tfjs-models/tree/master/body-pix#note-this-package-is-deprecated-in-favor-of-the-new-body-segmentation-package:~:text=Note%3A%20This%20package%20is%20deprecated%20in%20favor%20of%20the%20new%20body%2Dsegmentation%20package.). Thank you!

**Comment by satyamgaur** [comment]
Yeah I know it has been deprecated, but still it wasn't supposed to happen.. can you help me fixing it or what should I do to avoid this?

**Comment by gaikwadrahul8** [comment]
@satyamgaur 

As far I know body-pix is large model, you can check this [comment](https://github.com/tensorflow/tfjs/issues/2723#issuecomment-591671578) and that issue conversation about `body-pix `model so May know are you facing any issue with performance ? if yes then you can refer this [Tensorflow Blog](https://blog.tensorflow.org/2019/11/updated-bodypix-2.html) which may help you to solve your issue specifically this [section](https://blog.tensorflow.org/2019/11/updated-bodypix-2.html#:~:text=How%20to%20make,lower%20the%20accuracy.) about model size. Thank you!

**Comment by satyamgaur** [comment]
I went through all the links you mentioned @gaikwadrahul8, but couldn't find anything useful for our use case. Im basically looking for some sort of aggressive tree-shaking tool/way to only load what's absolutely required for our repo - to reduce the bundle size. I've read [this](https://github.com/tensorflow/tfjs-website/blob/master/docs/tutorials/deployment/size_optimized_bundles.md) for example, but couldn't make how to use this for the body-pix package, is this sort of modularisation available for body-pix? If not so, what can be done for now, and are you planning to add this in the future (in this or the replaced [body-segmentation package](https://github.com/tensorflow/tfjs-models/blob/master/body-segmentation))

**Comment by nileshau-afk** [comment]
Hello @satyamgaur  ,

Thank you for bringing this to our attention and for providing a comprehensive overview. We appreciate the time you invested in creating this issue.

Currently, our internal development team is concentrated on other scheduled milestones, so we cannot prioritize this task immediately. However, we would love to see this move forward as a community-led effort.

If you’d like to submit a fix, please fork the repo and open a PR following the [TensorFlow Contribution Guidelines](https://www.tensorflow.org/community/contribute), [Tfjs Contributions](https://github.com/tensorflow/tfjs/blob/master/tfjs-vis/CONTRIBUTING.md#how-to-contribute) and be sure to link this issue in your PR to avoid duplicate work, kindly leave a comment here if you plan to take this on.

Thank you again for your interest and time.

**Comment by nileshau-afk** [comment]
Hello @satyamgaur ,

Thank you for opening this issue. Since this issue has been open for a long time, we are closing this issue for now.
If this is still relevant and you would like to work on it, please feel free to reopen the issue or submit a pull request referencing it.

Thank you for your understanding.
